### PR TITLE
Correctly symlink tcpdf config when using the mono repository

### LINKED ISSUE #146: Failed to symlink tcpdf config
**Affected version(s)**
dev-master

**Description**  
The ScriptHandler::initializeAppliaction fails to symlink tcpdf.php. It looks for it in vendor\contao\core-bundle, which is obviously not there. 

```ERROR   system/config/tcpdf.php   Failed to create symbolic link from "C:\dev\web\contao-core\vendor\contao\core-bundle\src\Resources\contao\config\tcpdf.php" to "C:\dev\web\contao-core\system\config\tcpdf.php"```

**How to reproduce**  
1. Run ```composer create-project --no-install contao/managed-edition dev-master contao-core```
2. Adjust composer.json according to readme
3. Run ```composer update```

### PR DESCRIPTION
**Pull request by bezin**
This PR add adds a check for the monorepository to the SymlinksCommand and uses the correct path to symlink tcpdf config.

This fixed #146 .

### CODE REVIEW

File: core-bundle/src/Command/SymlinksCommand.php [hunk 1]
```diff
@@ -110,8 +111,11 @@ private function generateSymlinks(): void
         $this->symlink($this->getRelativePath($this->getContainer()->getParameter('kernel.logs_dir')), 'system/logs');
 
         // Symlink the TCPDF config file
-        $this->symlink('vendor/contao/core-bundle/src/Resources/contao/config/tcpdf.php', 'system/config/tcpdf.php');
-    }
+        if (isset($packages['contao/contao'])) {
+            $this->symlink('vendor/contao/contao/core-bundle/src/Resources/contao/config/tcpdf.php', 'system/config/tcpdf.php');
+        } else {
+            $this->symlink('vendor/contao/core-bundle/src/Resources/contao/config/tcpdf.php', 'system/config/tcpdf.php');
+        }    }
```
Comment (fritzmg) [inline core-bundle/src/Command/SymlinksCommand.php]: Some minor formatting issue here :)

Comment (bezin) [reply core-bundle/src/Command/SymlinksCommand.php]: Thanks for the hint, gonna fix this

**Comment by leofeyer** [comment]
I think the PR should be targeted at the `4.4` branch since it is a bugfix.

**Comment by bezin** [comment]
I thought about that as well... in 4.4 tcpdf.php is symlinked in the InstallCommand and was then moved to the SymlinksCommand. Should we backport this refactoring to 4.4 then? 

Besides: why is kernel.packages not existing in the tests? 🤔

**Comment by leofeyer** [comment]
> Should we backport this refactoring to 4.4 then?

No, in this case not. Then it needs to be targeted at the `4.6` branch.

> Besides: why is kernel.packages not existing in the tests?

Because we are using `vendor/composer/installed.json` to get the packages in Contao 4.4.

**Comment by bezin** [comment]
Ok, so as far as I see it:

1. Target this PR against ```4.5```  (since then the symlinking happens in SymlinkCommand). Changes can than be ported to 4.6
2. Fix in ```4.4``` as well. Do not use ```kernel.packages``` here, but ```vendor\composer\installed.json```

However I still don't get, why ```kernel.packages``` does not exist in the tests? Tests are run against ```master``` and this is the most recent code base, isn't it?

**Comment by leofeyer** [comment]
> Target this PR against `4.5`

Contao 4.5 has already reached its EOL. It is either `4.4` (LTS version), `4.6` (active version) or `master` (upcoming version).

**Review by leofeyer — CHANGES_REQUESTED** [review]
The PR needs to be retargeted at the `4.6` branch.

File: core-bundle/src/Command/SymlinksCommand.php (hunk 1 above)
Comment (dmolineus) [inline core-bundle/src/Command/SymlinksCommand.php]: You could use `kernel.bundles` and get the path of the `ContaoCoreBundle`. So you could get rid of the if statement, couldn't you?

**Comment by bezin** [comment]
See https://github.com/contao/contao/pull/158

**Comment by bezin** [comment]
Hello again,

just as I was about to use the mono repo for some 4.4 fixing, I remembered (or was remembered), that this fix is not yet backported to Contao 4.4. Hence, Contao 4.4 cannot be installed from the mono repository.

Quick recap:
 
The issue is the same as discussed here. The location of tcpdf.php config depends on the location of the Contao Core Bundle. We fixed this for Contao 4.6  in #158 

However, in Contao 4.4. the code is a little different: The symlinking of tcpdf.php is done in the InstallCommand rather than in the SymlinksCommand.

The question know is: Should we backport the refactoring from 4.6 to 4.4 or should we fix the issue in 4.4 as the code base is right now?

**Comment by bezin** [comment]
@contao/developers

**Comment by bezin** [comment]
@leofeyer @fritzmg @dmolineus

**Comment by leofeyer** [comment]
> Should we backport the refactoring from 4.6 to 4.4 or should we fix the issue in 4.4 as the code base is right now?

The latter.

**Pull request closed without merging**